### PR TITLE
Test pillow upd

### PR DESCRIPTION
**Pull request by aschroed**
1. Removed Pillow - wasn't being used effectively and frequently caused install issues.
    * Alex said that the image size dimensions are not used - though they could potentially be useful an approach using a CDN to generate thumbnails of a specified size and keep them cached could result in significant savings that would likely be more performant (to be explored)
    * The image.verify apparently was to ensure that the images were not corrupted or truncated but unclear if this really did anything or is necessary
2. Refactored the attachment code a bit 
    * tweaked the bit that checks a URL (either ftp or http) resolving to a file to raise Exceptions rather than exiting with sysexit.  
    * fixed a bug that would keep a non-ftp URL file on the users laptop after download
    * still uses magic to detect file types mime and still uses mimetypes to guess the mime based on the file extension and for all but zip files ensure that type and extension are consistent.
    * has a list of allowable mimetypes and raises an error if the guessed mime is not in the list
    * updated the tests for these changes
3. Updated setup to add long_description and change author_email and url.
4. **Will** requirements.txt and setup.py requires are not the same - I'm guessing they should be and not 100% on which things need pinned versions and which don't (only pytest-mock?) and where it is good practice to include the >= for versions.  Can you help resolve.
5. Signifcantly updated README 
6. Added functionality so 'tags' are treated in the same way as aliases in that you can append a tag to an item without overwriting any existing tags.

### CODE REVIEW

File: wranglertools/import_data.py [hunk 1]
```diff
@@ -154,65 +153,76 @@ def md5(path):
...
         if path.startswith("ftp://"):  # grab the file from ftp
             print("\nINFO: Attempting to download file from this url %s" % path)
-            with closing(urllib2.urlopen(path)) as r:
-                file_name = path.split("/")[-1]
-                with open(file_name, 'wb') as f:
-                    shutil.copyfileobj(r, f)
-                    path = file_name
-                    ftp_attach = True
+            try:
+                # import pdb; pdb.set_trace()
```
Comment (willronchetti) [inline wranglertools/import_data.py]: Delete

File: wranglertools/import_data.py [hunk 2]
```diff
@@ -154,65 +153,76 @@ def md5(path):
...
+                with closing(urllib2.urlopen(path)) as r:
+                    file_name = path.split("/")[-1]
+                    with open(file_name, 'wb') as f:
+                        shutil.copyfileobj(r, f)
+                        path = file_name
+                        ftp_attach = True
+            except urllib2.URLError as e:
+                raise Exception("\nERROR : FTP fetch for 'attachment' failed - {}".format(e))
```
Comment (willronchetti) [inline wranglertools/import_data.py]: I would recommend defining your own Exception type (and catching it in various tests if it meant to be caught). The below snippet will declare a named exception with no additional behavior. 
```
class MySuperSpecialException(Exception):
    """
    Named exception that you can throw like a normal one
    """
    pass
```

File: wranglertools/import_data.py [hunk 3]
```diff
@@ -154,65 +153,76 @@ def md5(path):
...
-        if not (minor == 'zip' or major == 'text'):  # zip files are special beasts
-            raise ValueError('Wrong extension for %s: %s' % (detected_type, filename))
+            try:
+                with open(path, "wb") as outfile:
+                    outfile.write(r.content)
+                    ftp_attach = True
+            except Exception as e:
+                raise Exception("\nERROR : Cannot write a tmp file to disk - {}".format(e))
+
+    allowed_mimes = (
```
Comment (willronchetti) [inline wranglertools/import_data.py]: I'd have this be a top level constant instead of a local variable

File: wranglertools/import_data.py [hunk 4]
```diff
@@ -154,65 +153,76 @@ def md5(path):
...
     attach = {}
+    filename = os.path.basename(path)
+    guessed_mime = mimetypes.guess_type(path)[0]
+    detected_mime = magic.from_file(path, mime=True)
+    # NOTE: this whole guesssing and detecting bit falls apart for zip files which seems a bit dodgy
+    # some .zip files are detected as generic application/octet-stream but don't see a good way to verify
+    # basically relying on extension with a little verification by magic for most file types
+    if detected_mime != guessed_mime and guessed_mime != 'application/zip':
+        raise ValueError('Wrong extension for %s: %s' % (detected_mime, filename))
+    if guessed_mime not in allowed_mimes:
```
Comment (willronchetti) [inline wranglertools/import_data.py]: I would check this before the above - the above is not worth checking if the below is not true, so check it first.

File: wranglertools/import_data.py (hunk 1 above)
Comment (aschroed) [reply wranglertools/import_data.py]: done

File: wranglertools/import_data.py (hunk 3 above)
Comment (aschroed) [reply wranglertools/import_data.py]: compromised to make it a constant within the function

**Pull request merged**